Composer/packages/client/src/components/ImportModal/ImportStatus.tsx

```diff
@@ -0,0 +1,167 @@
+  const { botName, source, state } = props;
+
+  const composerIcon = (
+    <img
+      alt={formatMessage('Composer Logo')}
+      aria-label={formatMessage('Composer Logo')}
+      src={compIcon}
+      style={{ width: '33px' }}
+    />
+  );
+
+  switch (state) {
+    case 'connecting': {
+      const label = (
+        <p style={{ fontSize: 16, whiteSpace: 'normal' }}>
+          {formatMessage('Connecting to ')}
+          <span css={boldBlueText}>{getUserFriendlySource(source)}</span>
+          {formatMessage(' to import bot content...')}
+        </p>
+      );
+      return (
+        <Dialog
+          dialogContentProps={contentProps}
+          hidden={false}
+          minWidth={560}
+          styles={{ main: { height: 263 } }}
+          modalProps={{ isBlocking: true }}
+        >
+          <span style={{ display: 'flex', justifyContent: 'center' }}>
+            {getServiceIcon(source)}
+            {composerIcon}
+          </span>
+          <ProgressIndicator label={label} styles={{ itemName: { textAlign: 'center' } }} />
+        </Dialog>
+      );
+    }
+
+    case 'downloading': {
+      const sourceName = getUserFriendlySource(source);
+      const label = (
+        <p style={{ fontSize: 16, whiteSpace: 'normal' }}>
+          {formatMessage('Importing ')}
+          <span css={boldText}>{botName}</span>
+          {formatMessage(` from ${sourceName}...`)}
+        </p>
+      );
+      return (
+        <Dialog
+          dialogContentProps={contentProps}
+          hidden={false}
+          minWidth={560}
+          styles={{ main: { height: 263 } }}
+          modalProps={{ isBlocking: true }}
+        >
+          <span style={{ display: 'flex', justifyContent: 'center' }}>
+            {getServiceIcon(source)}
+            {composerIcon}
+          </span>
+          <ProgressIndicator label={label} styles={{ itemName: { textAlign: 'center' } }} />
+        </Dialog>
+      );
+    }
+
+    default:
+      return <div style={{ display: 'none' }}></div>;
+  }
+};
+
+// TODO: typing of source
+function getServiceIcon(source?: string) {
+  let icon;
+  switch (source) {
+    case 'pva':
+      icon = (
+        <img
+          alt={formatMessage('PowerVirtualAgents Logo')}
+          aria-label={formatMessage('PowerVirtualAgents Logo')}
+          src={pvaIcon}
+          style={{ width: '33px' }}
+        />
+      );
+      break;
+
+    // don't draw anything, just render the Composer icon
+    default:
+      return undefined;
+  }
+  return (
+    <React.Fragment>
+      {icon}
+      <img
+        alt={formatMessage('Data transferring between services')}
+        aria-label={formatMessage('Data transferring between services')}
+        src={dataTransferLine}
+        style={{ margin: '0 16px', width: '78px' }}
+      />
+    </React.Fragment>
+  );
+}
+
+// TODO: create a type for possible publish sources
+export function getUserFriendlySource(source?: string): string {
```

Comment:
Why is this exported? Also please do define then use.

Reply:
This PR is a chunk of a larger feature branch where this function is reused. I'll remove the export for now